docs/source/dev-env.rst

```diff
@@ -0,0 +1,58 @@
+Zipline development guidelines
+==============================
+
+Commit messages
+---------------
+
+Standard acronyms to start the commit message with are:
+
+.. code-block:: bash
+
+   BLD: change related to building zipline
+   BUG: bug fix
+   DEP: deprecate something, or remove a deprecated object
+   DEV: development tool or utility
+   DOC: documentation
+   ENH: enhancement
+   MAINT: maintenance commit (refactoring, typos, etc.)
+   REV: revert an earlier commit
+   STY: style fix (whitespace, PEP8)
+   TST: addition or modification of tests
+   REL: related to releasing Zipline
+   PERF: Performance enhancements
+
+Some commit style guidelines:
+
+Commit lines should be no longer than 72 characters. First line of the commit should include a the aforementioned prefix. There should be an empty line between the commit subject and the body of the commit. In general, the message should be in the imperative tense. Best practice is to include not only what change, by why.
+
```

Comment:
Small typos/grammar suggestions: 

- "but why" rather than "by why"
- "commit should include one of the above prefixes"